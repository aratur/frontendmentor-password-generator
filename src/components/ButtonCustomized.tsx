import React from 'react';
import Button, { ButtonProps } from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import arrowImg from '../assets/images/icon-arrow-right.svg';
import useMediaHook from '../hooks/mediaHook';

interface Props extends ButtonProps {
  onClick: () => void;
  name: string;
}

const ButtonCustomized = (props: Props) => {
  const { onClick, name, ...otherProps } = props;
  const media = useMediaHook();
  return (
    <Button
      onClick={onClick}
      endIcon={
        <img className="button__icon--hover" src={arrowImg} alt="arrow" />
      }
      sx={{
        backgroundColor: 'primary.main',
        borderRadius: 0,
        height: media ? '65px' : '56px',
        border: '1px solid #a4ffaf',
        color: 'primary.dark',
        img: {
          transition: 'filter 200ms ease-in ',
          filter:
            'invert(12%) sepia(21%) saturate(430%) hue-rotate(207deg) brightness(97%) contrast(96%)',
        },
        '&:hover, &:focus ': {
          color: 'primary.main',
          backgroundColor: 'primary.dark',
          img: {
            filter:
              'invert(81%) sepia(33%) saturate(409%) hue-rotate(75deg) brightness(105%) contrast(103%)',
          },
        },
      }}
      {...otherProps}
    >
      <Typography variant={media ? 'body2' : 'h3'} component="span">
        {name}
      </Typography>
    </Button>
  );
};

export default ButtonCustomized;
