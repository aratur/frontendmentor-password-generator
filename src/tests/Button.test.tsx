import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import ButtonCustomized from '../components/ButtonCustomized';

describe('ButtonCustomized', () => {
  it('snapshot', () => {
    const handleClicked = vi.fn();
    const { container } = render(
      <ButtonCustomized handleClicked={handleClicked} />
    );
    expect(container).toMatchSnapshot();
  });
  it('calls prop function when clicked', () => {
    const handleClicked = vi.fn();
    render(<ButtonCustomized handleClicked={handleClicked} />);
    const button = screen.getByRole('button');
    fireEvent.click(button);
    expect(handleClicked).toHaveBeenCalled();
  });
});
