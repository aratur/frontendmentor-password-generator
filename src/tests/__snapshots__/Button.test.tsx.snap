// Vitest Snapshot v1

exports[`ButtonCustomized > snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-19bag00-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
    >
      Generate
    </span>
    <span
      class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
    >
      <img
        alt="arrow"
        class="button__icon_hover"
        src="/src/assets/images/icon-arrow-right.svg"
      />
    </span>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
